### PR TITLE
Create fmf for each tcms case when converting.

### PR DESCRIPTION
**Pull request by hegerj**
Using tmt test convert will now create fmf file for each test case found
in tcms apart from main.fmf, which contains common attributes found in
all of the individual test cases. New files are named according to their
tcms id and contain only attributes with values that are unique to it.

### CODE REVIEW

**Comment by hegerj** [comment]
I wasn't sure where to document the changed behaviour so I would appreciate an advice.

**Comment by psss** [comment]
Tests are failing:

    <Result UnboundLocalError("local variable 'testcase_data' referenced before assignment")>.exit_code

> I wasn't sure where to document the changed behaviour so I would appreciate an advice.

Documentation updates should go here:

* https://tmt.readthedocs.io/en/latest/examples.html#convert-tests

You can find the source code under `docs/examples.rst`. To review updated docs locally:

    pip install -e .[docs]
    make docs
    firefox docs/_build/html/index.html

Thanks.

File: tmt/convert.py [hunk 1]
```diff
@@ -109,47 +110,80 @@ def read(path, makefile, nitrate, purpose):
...
-                testcase.arguments)
-            echo(style('environment:', fg='green'))
-            echo(pprint.pformat(data['environment']))
-        # Relevancy
-        field = tmt.utils.StructuredField(testcase.notes)
-        data['relevancy'] = field.get('relevancy')
-        echo(style('relevancy:', fg='green'))
-        echo(data['relevancy'].rstrip('\n'))
+            log.warn("Multiple test cases found for '{0}'.".format(
+                test))
```
Comment (psss) [inline tmt/convert.py]: Shall we still give warning when multiple test cases found? Or just give and info message?

File: tmt/convert.py [hunk 2]
```diff
@@ -109,47 +110,80 @@ def read(path, makefile, nitrate, purpose):
...
+
+        testcase_data = list()
+
+        for testcase in testcases:
+            single_case_data = dict()
```
Comment (psss) [inline tmt/convert.py]: Seems the `read()` function is getting a little bit too large. Would it make sense to create a separate function `read_tmcs()` and call it from here?

File: tmt/convert.py [hunk 3]
```diff
@@ -109,47 +110,80 @@ def read(path, makefile, nitrate, purpose):
...
+                    if key in common_candidates:
+                        if value != common_candidates[key]:
+                            common_candidates.pop(key)
+
+        # Add common data to main.fmf
+        for key, value in common_candidates.items():
+            data[key] = value
+
+        # Remove common data from individual fmfs
+        for common_key in list(common_candidates):
```
Comment (psss) [inline tmt/convert.py]: I guess this could be just `for common_key in common_candidates`, right?

File: tmt/convert.py (hunk 2 above)
Comment (hegerj) [reply tmt/convert.py]: In theory yes, however I don't really see someone using it other than `read()` as it is pretty dependant on it.

File: tmt/convert.py (hunk 1 above)
Comment (hegerj) [reply tmt/convert.py]: info seems better

File: tmt/convert.py (hunk 3 above)
Comment (hegerj) [reply tmt/convert.py]: I was convinced that without the `list()` it caused some kind of problem but I cannot replicate it now so I guess it is pointless.

File: tmt/convert.py (hunk 1 above)
Comment (hegerj) [reply tmt/convert.py]: Although `logger.info()` only displays with higher level of `DEBUG`.
Is there a logger or other function which would display by default or do you think this information shouldn't be seen unless you are running higherr `DEBUG`?

Comment (psss) [reply tmt/convert.py]: Under `tmt run` we're using the method `info()` for such output. We could somehow enable it here as well. See also `pydoc tmt.utils.Common` for more info about the common class.

Comment (psss) [reply tmt/convert.py]: Seems that it depends on the workdir functionality so it would need some adjustments...

**Comment by psss** [comment]
Looks good now. Thanks. I see a couple of minor issues. We can improve them later.

**Pull request merged**